src/Files.App/Helpers/DynamicDialogFactory.cs

```diff
@@ -103,6 +103,18 @@ public static DynamicDialog GetFor_RenameDialog()
 				}
 			};
 
+			inputText.TextChanged += (textBox, args) =>
+			{
+				if (!string.IsNullOrWhiteSpace(inputText.Text))
+				{
+					dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
+				}
+				else
+				{
+					dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
+				}
```

Comment:
Can be simplified (here without the proper indentation):  
```suggestion
dialog.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) 
    ? DynamicDialogButtons.Cancel 
    : DynamicDialogButtons.Cancel | DynamicDialogButtons.Primary ;
```